src/scripts/charts-screenshots.ts

```diff
@@ -1,100 +1,96 @@
-  return sizeInBytes;
+        if (totalHeight >= scrollHeight) {
+          clearInterval(timer);
+          resolve('success');
+        }
+      }, 100);
+    });
+  });
 }
 
 async function updateChartScreenshots(): Promise<void> {
-  const browser = await puppeteer.launch({ headless: true });
-  const page = await browser.newPage();
-  // await page.emulate(iPad);
-  await page.setViewport({ width: 1200, height: 1000 });
-  let i = 0;
-  const pageLength = chartUrls.length;
-  for (; i < pageLength; i++) {
-    const fileNameSave = `${chartUrls[i].split('/').pop()}.png`;
+  try {
+    const browser = await puppeteer.launch({ headless: true });
+    const page = await browser.newPage();
+
+    await page.setViewport({ width: 1200, height: 800 });
 
-    await page.goto(`${FRONTEND_SITE_URL}${chartUrls[i]}`);
-    await delay(3000);
-    const [button] = await page.$x(
-      "//button[contains(text(), 'Download PNG')]",
-    );
-    let saved = false;
-    // get fist file
-    // let saved = false;
-    page.on('response', async response => {
-      let url = response.request().url();
-      const contentType = response.headers()['content-type'];
+    const pageLength = chartUrls.length;
+    try {
+      for (let i = 0; i < pageLength; i++) {
+        const timeStart = new Date().getTime();
+        const fileNameSave = `${chartUrls[i].split('/').pop()}.png`;
+        const client = await page.target().createCDPSession();
+        await page.goto(`${FRONTEND_SITE_URL}${chartUrls[i]}`);
 
-      if (
-        !saved &&
-        contentType &&
-        contentType.endsWith('image/png') &&
-        url.startsWith('data:image/png;base64,')
-      ) {
-        const fileSize = getFileSize(url);
-        console.log({ fileSize });
-        if (fileSize > 15) {
-          saved = true;
-          url = url.replace('data:image/png;base64,', '');
-          fs.writeFile(
-            `src/public/charts/${fileNameSave}`,
-            url,
-            'base64',
-            function (err) {
-              if (err) {
-                console.error(err);
-                return;
+        // Waiting the chart component loaded
+        await page.waitForSelector('.echarts-for-react');
+
+        const [button] = await page.$x(
+          "//button[contains(text(), 'Download PNG')]",
+        );
+        const folder = path.join(
+          __dirname,
+          '../../public/charts',
+          chartUrls[i].split('/').pop(),
+        );
+        await client.send('Page.setDownloadBehavior', {
+          behavior: 'allow',
+          downloadPath: folder,
+        });
+        // scroll to the bottom to download the chart
+        await autoScroll(page);
+        // waiting for the scroll to the bottom and the chart animation
+        if (button) {
+          await button.click();
+          // await the file downloaded
+          await page.waitForTimeout(3000);
+
+          if (fs.existsSync(folder)) {
```

Comment:
`existsSync` and `readdirSync` are blocking operations which may block entire express server, please change to await

Reply:
Ok. Thanks.
I just updated
https://github.com/pastelnetwork/pastel-explorer-backend/pull/45/commits/c3084d315a3df37b113a439cae6c7608bc966c56